network/struct.go

```diff
@@ -107,6 +109,7 @@ func NewServerIdentity(public abstract.Point, address Address) *ServerIdentity {
 		Public:  public,
 		Address: address,
 		ID:      ServerIdentityID(uuid.NewV5(uuid.NamespaceURL, url)),
+		Cert:    []byte{},
```

Comment:
Ok I've looked into protobuf and if you define the field as : 
``` type myStruct struct {
...
Cert []byte `protobuf:"opt"`
...
}
```
It becomes optional so protobuf does not complain anymore.
It makes much sense since TLS is *still* optional !
The thing with having a `NewTLSServerIdentity` is it would avoid doing all these assigments `si.Cert = cert` right after creating a `ServerIdentity`. I'm proposing one way to get rid of that, if you find another one that's also good.